src/amazonica/aws/kinesis.clj

```diff
@@ -123,26 +123,37 @@
       (Thread/sleep 3000)
       false)))
 
+(def ^:dynamic *checkpointer*
+  "Bound to an IRecordProcessorCheckpointer when using manual checkpointing.")
+
 (defn- processor-factory
-  [processor deserializer checkpoint next-check]
+  [processor deserializer checkpoint-strategy checkpoint-timeout next-check]
   (reify IRecordProcessorFactory
     (createProcessor [this]
       (reify IRecordProcessor
         (initialize [this shard-id])
         (shutdown [this checkpointer reason]
-          (if (or (= ShutdownReason/TERMINATE reason)
-                  (= "TERMINATE" reason))
+          (if (and (or (= ShutdownReason/TERMINATE reason)
+                       (= "TERMINATE" reason))
+                   ;; those using manual checkpointing need to decide
+                   ;; whether to checkpoint during shutdown on their own.
+                   (not= checkpoint-strategy :manual))
               (some (partial mark-checkpoint checkpointer) [1 2 3 4 5])))
         (processRecords [this records checkpointer]
-          (if (or (processor (functor/fmap (partial marshall deserializer)
-                                           (vec (seq records))))
-                  (and checkpoint
-                       (> (System/currentTimeMillis) @next-check)))
-              (do (if checkpoint
-                      (reset! next-check
-                              (+' (System/currentTimeMillis)
-                                  (*' 1000 checkpoint))))
-                  (some (partial mark-checkpoint checkpointer) [1 2 3 4 5]))))))))
+          (if (= checkpoint-strategy :manual)
+            (binding [*checkpointer* checkpointer]
```

Comment:
What's bound to `*checkpointer*` when the `checkpoint-strategy` is not `:manual`?